File: backend/app/schemas/workspace.py
from pydantic import BaseModel, Field
from datetime import datetime
from typing import Optional, Dict, Any


class WorkspaceBase(BaseModel):
    """Base workspace schema."""
    name: str = Field(..., min_length=1, max_length=255)
    description: Optional[str] = None
    settings: Optional[Dict[str, Any]] = None


class WorkspaceCreate(WorkspaceBase):
    """Schema for creating a workspace."""
    owner_id: str


class WorkspaceUpdate(BaseModel):
    """Schema for updating a workspace."""
    name: Optional[str] = Field(None, min_length=1, max_length=255)
    description: Optional[str] = None
    settings: Optional[Dict[str, Any]] = None


class WorkspaceResponse(WorkspaceBase):
    """Schema for workspace response."""
    id: str
    owner_id: str
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/workspace.ts
export interface Workspace {
  id: string;
  name: string;
  description?: string;
  owner_id: string;
  created_at: string;
  updated_at: string;
  settings?: Record<string, any>;
}

export interface WorkspaceCreate {
  name: string;
  description?: string;
  owner_id: string;
  settings?: Record<string, any>;
}

export interface WorkspaceUpdate {
  name?: string;
  description?: string;
  settings?: Record<string, any>;
}
